sessiontxn/txn_context_test.go

```diff
@@ -882,3 +883,91 @@ func TestOptimisticTxnRetryInPessimisticMode(t *testing.T) {
 		}
 	})
 }
+
+func TestTSOCmdCountForPrepareExecute(t *testing.T) {
+	// This is a mock workload mocks one which discovers that the tso request count is abnormal.
+	// After the bug fix, the tso request count recovers, so we use this workload to record the current tso request count
+	// to reject future works that accidentally causes tso request increasing.
+	// Note, we do not record all tso requests but some typical requests.
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/sessiontxn/isolation/requestTsoFromPD", "return"))
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+
+	ctx := context.Background()
+	tk := testkit.NewTestKit(t, store)
+
+	sctx := tk.Session()
+	sctx.SetValue(sessiontxn.TsoRequestCount, 0)
```

Comment:
Why not set the `TsoRequestCount` before loop, `117` in `require.Equal(t, uint64(117), count)` is a little harder to understand